test/cppsuite/tests/search_near_01.cxx

```diff
@@ -0,0 +1,249 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "test_harness/test.h"
+#include "test_harness/workload/thread_context.h"
+#include "test_harness/thread_manager.h"
+#include "test_harness/workload/random_generator.h"
+#include "test_harness/util/api_const.h"
+
+using namespace test_harness;
+/*
+ * In this test, we want to verify that search_near with prefix enabled only traverses the portion
+ * of the tree that follows the prefix portion of the search key. The test is composed of a populate
+ * phase followed by a read phase. The populate phase will insert a set of random generated keys
+ * with a prefix of aaa -> zzz. The read phase will continously perform prefix search near calls,
+ * and validate that the number of entries traversed is within bounds of the search key.
+ */
+class search_near_01 : public test_harness::test {
+    uint64_t keys_per_prefix = 0;
+    uint32_t srchkey_len = 0;
+    std::string alphabet{"abcdefghijklmnopqrstuvwxyz"};
+    const uint32_t ALPHABET_SIZE = 26;
+    const uint32_t PREFIX_KEY_LEN = 3;
+
+    int64_t
+    get_stat(test_harness::thread_context *tc, int stat_field)
```

Comment:
Could we use the one from `runtime_monitor` somehow instead of duplicating the code here?

Reply:
We could potentially make the `get_stat` function in runtime_monitor a static function. Do you think we should do that here or WT-8092?

Reply:
You can do it here as it's needed by that test only.